File: room.js
import { model, Schema } from "mongoose";

const roomSchema = new Schema({
  roomName: {
    type: String,
    required: true,
  },

  users: [
    {
      userToken: {
        type: String,
        required: true,
      },
      userName: {
        type: String,
      },
      score: {
        type: Number,
        default: 0,
      },
      checkedItems: {
        type: Schema.Types.Array,
      },
    },
  ],

  items: {
    type: Schema.Types.Array,
  },
});

export const Room = model("room", roomSchema);

File: Vision_DB/DumplingChaseDataStore/Room.cs
namespace DumplingChaseDataStore
{
    public class Room
    {
        public string RoomName { get; set; }
        public List<User> Users { get; set; } = new();
        public List<string> Items { get; set; } = new();

        private static Room _emptyRoom = new Room();
        public static Room Empty
        {
            get
            {
                return _emptyRoom;
            }
        }
    }
}